app/src/main/java/com/android/sample/ui/activity/CreateActivity.kt

```diff
@@ -554,7 +555,15 @@ fun CreateActivityScreen(
                           dueDate.toDate().after(Timestamp.now().toDate()),
                   onClick = {
                     val activityId = listActivityViewModel.getNewUid()
-                    if (creator == "") {
+                    // we don't want to allow creators to create activities 1 hour before they start
+                    if (dueDate.toDate().time - System.currentTimeMillis() <
+                        TimeUnit.HOURS.toMillis(1)) {
+                      Toast.makeText(
+                              context,
+                              "Activity must be scheduled at least 1 hour in advance",
```

Comment:
Please put this text in our C resource file.

Reply:
I think you meant string.xml. I have corrected this in new version